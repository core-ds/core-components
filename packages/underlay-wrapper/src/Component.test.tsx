import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { UnderlayWrapper } from './index';

describe('UnderlayWrapper', () => {
    describe('Snapshots tests', () => {
        it('should match snapshot', () => {
            const { container } = render(
                <UnderlayWrapper
                    borderSize={2}
                    backgroundColor='info'
                    borderColor='graphic-link'
                    shadow='shadow-l'
                >
                    Content
                </UnderlayWrapper>,
            );

            expect(container).toMatchSnapshot();
        });
    });

    it('should set `data-test-id` attribute', () => {
        const dataTestId = 'test-id';
        const { getByTestId } = render(<UnderlayWrapper dataTestId={dataTestId} />);

        expect(getByTestId(dataTestId).tagName).toBe('DIV');
    });

    describe('Classes tests', () => {
        it('should set `className` class', () => {
            const className = 'test-class';
            const dataTestId = 'test-id';
            const { getByTestId } = render(
                <UnderlayWrapper className={className} dataTestId={dataTestId} />,
            );

            expect(getByTestId(dataTestId)).toHaveClass(className);
        });

        it('should set `background-accent` class if `backgroundColor` prop is `accent`', () => {
            const backgroundColor = 'accent';
            const dataTestId = 'test-id';
            const { getByTestId } = render(
                <UnderlayWrapper backgroundColor={backgroundColor} dataTestId={dataTestId} />,
            );

            expect(getByTestId(dataTestId)).toHaveClass(`background-${backgroundColor}`);
        });

        it('should set `border-color-accent` class if `borderColor` prop is `accent`', () => {
            const borderColor = 'accent';
            const dataTestId = 'test-id';
            const { getByTestId } = render(
                <UnderlayWrapper borderColor={borderColor} dataTestId={dataTestId} />,
            );

            expect(getByTestId(dataTestId)).toHaveClass(`border-color-${borderColor}`);
        });

        it('should set `shadow-xs` class if `shadow` prop is `shadow-xs`', () => {
            const shadow = 'shadow-xs';
            const dataTestId = 'test-id';
            const { getByTestId } = render(
                <UnderlayWrapper shadow={shadow} dataTestId={dataTestId} />,
            );

            expect(getByTestId(dataTestId)).toHaveClass(shadow);
        });

        it('should set `border-width-1` class if `borderSize` prop is `1`', () => {
            const borderSize = 1;
            const dataTestId = 'test-id';
            const { getByTestId } = render(
                <UnderlayWrapper borderSize={borderSize} dataTestId={dataTestId} />,
            );

            expect(getByTestId(dataTestId)).toHaveClass(`border-width-${borderSize}`);
        });
    });

    describe('Callbacks tests', () => {
        it('should call `onClick` prop', async () => {
            const cb = jest.fn();
            const dataTestId = 'test-id';
            const { getByTestId } = render(
                <UnderlayWrapper onClick={cb} dataTestId={dataTestId} />,
            );

            const el = getByTestId(dataTestId);

            fireEvent.click(el);

            expect(cb).toBeCalledTimes(1);
        });
    });

    it('should unmount without errors', () => {
        const { unmount } = render(<UnderlayWrapper />);

        expect(unmount).not.toThrowError();
    });
});
