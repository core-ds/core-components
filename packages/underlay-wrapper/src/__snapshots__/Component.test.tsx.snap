// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnderlayWrapper Snapshots tests should match snapshot 1`] = `
<div>
  <div
    class="component info graphic-link border-width-2 hadow-l"
    role="button"
    style="width: 200px; height: 200px; box-shadow: hadow-l;"
    tabindex="0"
  >
    Content
  </div>
</div>
`;
